#! /bin/sh /usr/share/dpatch/dpatch-run
## 01-nv_list_memory_problem.dpatch by Sebastien Delafond <seb@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix an out-of-memory error

@DPATCH@
diff -urNad zfs-fuse-0.6.0~/src/zfs-fuse/zfs_ioctl.c zfs-fuse-0.6.0/src/zfs-fuse/zfs_ioctl.c
--- zfs-fuse-0.6.0~/src/zfs-fuse/zfs_ioctl.c	2009-08-16 04:52:22.000000000 +0200
+++ zfs-fuse-0.6.0/src/zfs-fuse/zfs_ioctl.c	2009-12-07 20:37:37.000000000 +0100
@@ -834,11 +834,7 @@
 		  syslog(LOG_WARNING,"put_nvlist: error %s on xcopyout",strerror(error));
 	}
 
-	/* zc->zc_nvlist_dst_size = size; */
-	/* This commented allocation was probably some kind of optimization
-	since this zc is sent to the socket. Except that put_nvlist is sometimes
-	called recursively and in this case we get very fast an out of memory error
-	in this function. Simply commenting out the allocation fixes the problem */
+	zc->zc_nvlist_dst_size = size; 
 	return (error);
 }
 
